File: AbyssalCreatures/src/main/java/com/offz/spigot/abyssalcreatures/mobs/hostile/Tamaugachi.java
/*
package com.offz.spigot.abyssalcreatures.mobs.hostile;

import com.offz.spigot.mobzy.mobs.behaviours.HitBehaviour;
import com.offz.spigot.mobzy.mobs.types.HostileMob;
import com.offz.spigot.mobzy.pathfinders.PathfinderGoalWalkingAnimation;
import net.minecraft.server.v1_15_R1.World;
import org.bukkit.Material;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.LivingEntity;
import org.bukkit.inventory.EntityEquipment;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

public class Tamaugachi extends HostileMob implements HitBehaviour {
    public Tamaugachi(World world) {
        super(world, "Tamaugachi");
        this.setSize(2F, 2.5F);

        EntityEquipment ee = ((LivingEntity) getBukkitEntity()).getEquipment();
        ItemStack is = new ItemStack(Material.STONE);
        ItemMeta itemMeta = is.getItemMeta();
        itemMeta.setUnbreakable(true);
        itemMeta.addEnchant(Enchantment.DEPTH_STRIDER, 40, true);
        is.setItemMeta(itemMeta);
        ee.setBoots(is);
    }

    @Override
    public void createPathfinders() {
        super.createPathfinders();
        goalSelector.a(0, new PathfinderGoalWalkingAnimation(this, getStaticBuilder().getModelID()));
    }
}*/

File: AbyssalCreatures/src/main/java/com/offz/spigot/abyssalcreatures/mobs/hostile/Tamaugachi.kt
package com.offz.spigot.abyssalcreatures.mobs.hostile

import com.mineinabyss.idofront.items.editItemMeta
import com.offz.spigot.mobzy.mobs.behaviours.HitBehaviour
import com.offz.spigot.mobzy.mobs.types.HostileMob
import net.minecraft.server.v1_15_R1.World
import org.bukkit.Material
import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.ItemStack

class Tamaugachi(world: World?) : HostileMob(world, "Tamaugachi"), HitBehaviour {
    init {
        //make them walk fast in water thanks to depth strider. The material can't be boots since they'll display even
        // if the entity is invisible
        living.equipment!!.boots = ItemStack(Material.STONE).editItemMeta {
            it.isUnbreakable = true
            it.addEnchant(Enchantment.DEPTH_STRIDER, 40, true)
        }
    }
}